enhancements/oc/day2-add-nodes.md

```diff
@@ -0,0 +1,264 @@
+
+The following steps illustrates the expected workflow:
+1. The cluster administrator prepares the `nodes-config.yaml` file listing all the nodes to be added, optionally with their initial network configuration
+2. The cluster administrator runs the `oc adm add-nodes-image` command to generate the ISO
+3. The cluster administrator boots every node previously listed using the freshly generated ISO
+4. The cluster administrator runs the `oc adm monitor-add-nodes` command to monitor the activity progress
+5. Once a node csr is pending, the cluster administrator runs the `oc adm certificate approve <csr-id>` to complete the process for the specified node
+
+### API Extensions
+N/A
+
+### Implementation Details/Notes/Constraints
+The implementation will largely reuse the same approach / code present in the openshift/installer repo for creating an ISO image 
+for the installation, with a number of exceptions that will be detailed in the subsequent paragraphs.
+
+#### Proposed architecture
+Every node booted with the generated ISO will will be independent for adding itself to the cluster - without any need to specify a single rendezvous IP, 
+as it happens for the installation scenario. In this way if a node is failing for some reason (for example, the pre-flight validations block the process), 
+the other nodes may continue the joining process undisturbed.
+The following hilights better the proposed architecture:
+
+![alt Architecture diagram](day2-add-nodes-diagram.png)
+
+#### Assisted-service interaction
+The following API calls will be used to add a single node to the target cluster:
+- `/api/assisted-install/v2/clusters/import` to import an existing cluster
+- `/api/assisted-install/infra-envs/${INFRA_ENV_ID}/hosts/${HOST_ID}` to add the specified host to the cluster
+
+#### oc integration notes
+Since most of the existing code for agent-based installer approach is currently stored into the openshift/installer repo, the proposal is to implement the 
+commands in a way similar to the already existing `oc adm must-gather` command, where essentially a pod is run within the target cluster to execute specific 
+code build from the installer and shipped into a release image. Such approach will offer the following benefits:
+
+- No code duplication/pollution. The required code for the additional commands will exist only in the openshift/installer repo (it will not be vendored into 
+  the openshift/oc repo)
+- Simplified integration. `oc` will take care only of marshaling the inputs/outputs of the command
+
+To achieve this kind of loosely coupled level of integration with `oc`, a specific binary with the described commands will be built from the openshift/installer 
+repo code and distributed into an image of the release payload as an additional artifact.
+
+### Risks and Mitigations
+Most of the existing code for agent-based installer approach is currently stored into the openshift/installer repo (with additional vendoring of the api/client 
```

Comment:
We are struggling with the maintenance issues(security scanners, code checkers, bugs, etc.) stemming from the vendored packages in oc and I think, we should be very reluctant to add more(installer, assisted-service). Since we already vendor openshift/api, this package can work.

Reply:
This text is in the wrong section I think. It should be in the Alternatives section because vendoring all of the installer code into `oc` is something that we are explicitly not proposing to do.

`oc`'s job here (as described in the "oc integration notes" section above) is going to be to start a pod (a bit like the `must-gather` command) and stream out the ISO (again like `must-gather`) and the installer progress (like `oc logs`). It's syntactic sugar for stuff that `oc` is can already do.

@andfasano can you move this into the 'Alternatives' section to make it clear that we are _not_ planning to vendor a bunch of stuff into oc?

Reply:
Thanks for the reply. Could you please add that which packages would be possibly needed to vendor in oc to see what are their dependency chain depths?.

Reply:
Agree with @zaneb , the section is not the proper one, will move it

Reply:
Thanks for the clarification https://github.com/openshift/enhancements/pull/1517/files#diff-69aa9ff9e2b01a83a9d639863363dc17b351e16792d74d580142c0764056b6d4R205-R206

Reply:
> Could you please add that which packages would be possibly needed to vendor in oc to see what are their dependency chain depths?.

I don't anticipate there being any at all.
It'll be purely doing stuff that oc can already do, like creating a Pod and streaming the output to a file/the console.